internal/ansible/handler/logging_enqueue_annotation.go

```diff
@@ -0,0 +1,112 @@
+// limitations under the License.
+package handler
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/operator-framework/operator-lib/handler"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/util/workqueue"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+)
+
+const (
+	nilString = "<nil>"
+)
+
+// LoggingEnqueueRequestForAnnotation wraps operator-lib handler for
+// "InstrumentedEnqueueRequestForObject", and logs the events as they occur
+//		&handler.LoggingEnqueueRequestForAnnotation{}
+type LoggingEnqueueRequestForAnnotation struct {
+	handler.EnqueueRequestForAnnotation
+}
+
+// Create implements EventHandler, and emits a log message.
+func (h LoggingEnqueueRequestForAnnotation) Create(e event.CreateEvent, q workqueue.RateLimitingInterface) {
+	h.logEvent("Create", e.Object, nil)
+	h.EnqueueRequestForAnnotation.Create(e, q)
+}
+
+// Update implements EventHandler, and emits a log message.
+func (h LoggingEnqueueRequestForAnnotation) Update(e event.UpdateEvent, q workqueue.RateLimitingInterface) {
+	h.logEvent("Update", e.ObjectOld, e.ObjectNew)
+	h.EnqueueRequestForAnnotation.Update(e, q)
+}
+
+// Delete implements EventHandler, and emits a log message.
+func (h LoggingEnqueueRequestForAnnotation) Delete(e event.DeleteEvent, q workqueue.RateLimitingInterface) {
+	h.logEvent("Delete", e.Object, nil)
+	h.EnqueueRequestForAnnotation.Delete(e, q)
+}
+
+// Generic implements EventHandler, and emits a log message.
+func (h LoggingEnqueueRequestForAnnotation) Generic(e event.GenericEvent, q workqueue.RateLimitingInterface) {
+	h.logEvent("Generic", e.Object, nil)
+	h.EnqueueRequestForAnnotation.Generic(e, q)
+}
+
+func (h LoggingEnqueueRequestForAnnotation) logEvent(eventType string, object, newObject client.Object) {
+	typeString, name, namespace := extractTypedOwnerAnnotations(h.EnqueueRequestForAnnotation.Type, object)
+	if newObject != nil && typeString == "" {
+		typeString, name, namespace = extractTypedOwnerAnnotations(h.EnqueueRequestForAnnotation.Type, newObject)
+	}
+	if namespace == "" {
+		namespace = nilString
+	}
+
+	objectNs := object.GetNamespace()
+	if objectNs == "" {
+		objectNs = nilString
+	}
+
+	if name != "" && typeString != "" {
+		log.Info(fmt.Sprintf("Received %s event for dependent resource with GVK %s with name %s in namespace %s, mapped to owner GK %s with name %s in namespace %s",
+			eventType,
+			object.GetObjectKind().GroupVersionKind().String(),
+			object.GetName(),
+			objectNs,
+			typeString,
+			name,
+			namespace))
+	}
+}
+
+func extractTypedOwnerAnnotations(ownerGK schema.GroupKind, object metav1.Object) (string, string, string) {
+	if typeString, ok := object.GetAnnotations()[handler.TypeAnnotation]; ok && typeString == ownerGK.String() {
+		if namespacedNameString, ok := object.GetAnnotations()[handler.NamespacedNameAnnotation]; ok {
```

Comment:
Nit: annotations could be nil, and readability
```suggestion
	annotations := object.GetAnnotations()
	if len(annotations) == 0 {
		return "", "", ""
	}
	if typeString, ok := annotations[handler.TypeAnnotation]; ok && typeString == ownerGK.String() {
		if namespacedNameString, ok := annotations[handler.NamespacedNameAnnotation]; ok {
```

Reply:
Also, is it ok to return `typeString` if no namespacedname annotation exists?

Reply:
yeah, if at least typeString and name aren't returned we'll just skip the log message anyway https://github.com/operator-framework/operator-sdk/pull/4779/files#diff-0718f6160c2f87214f996ac463b60e21708b2faafff39c930160837757f185bdR66